sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -392,4 +392,278 @@ case class GpuArrayMax(child: Expression) extends GpuBaseArrayAgg with ImplicitC
   override def prettyName: String = "array_max"
 
   override protected def agg: GroupByAggregation = GroupByAggregation.max()
-}
\ No newline at end of file
+}
+
+class GpuSequenceMeta(
+    expr: Sequence,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends ExprMeta[Sequence](expr, conf, parent, rule) {
+
+  override def tagExprForGpu(): Unit = {
+    //  We have to fall back to the CPU if the timeZoneId is not UTC when
+    //  we are processing date/timestamp.
+    //  Date/Timestamp are not enabled right now so this is probably fine.
+  }
+
+  override def convertToGpu(): GpuExpression = {
+    if (expr.stepOpt.isDefined) {
+      val Seq(start, stop, step) = childExprs.map(_.convertToGpu())
+      GpuSequenceWithStep(start, stop, step, expr.timeZoneId)
+    } else {
+      val Seq(start, stop) = childExprs.map(_.convertToGpu())
+      GpuSequence(start, stop, expr.timeZoneId)
+    }
+  }
+}
+
+trait GpuSequenceTrait {
```

Comment:
nit: Why is this a trait and not an object?

Reply:
Done. Thx